reference/constraints/Traverse.rst

```diff
@@ -37,31 +34,66 @@ configure the ``Traverse`` constraint on the ``Book`` class.
          * @ORM\Entity
          * @Assert\Traverse
          */
-        class Book
+        class BookCollection implements \IteratorAggregate
         {
             /**
-             * @var Author
+             * @var string
+             *
+             * @ORM\Column
              *
-             * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Author")
+             * @Assert\NotBlank
              */
-            protected $author;
+            protected $name = '';
 
             /**
-             * @var Editor
+             * @var Book[]
              *
-             * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Editor")
+             * @ORM\ManyToMany(targetEntity="AppBundle\Entity\Book")
              */
-            protected $editor;
+            protected $books;
 
-            // ...
+            // some other properties
+
+            public function __construct()
+            {
+                $this->books = new ArrayCollection();
+            }
+
+            // ... setter for name, adder and remover for books
+
+            // the name can be validated by calling the getter
+            public function getName(): string
+            {
+                return $this->name;
+            }
+
+            /**
+             * @return Book[]|\Generator The books for a given author
```

Comment:
```suggestion
             * @return \Generator|Book[] The books for a given author
```
(real type first)
(actually it ought to be `\Generator<Book>` if we had generics...)